File: leetcode/59.螺旋矩阵 II/59.java
class Solution {
    public int[][] generateMatrix(int n) {
        int[][] maxtrix = new int[n][n];
        int cnt = 1;
        int left = 0, right = n - 1, top = 0, buttom = n -1;
        while(left <= right && top <= buttom && cnt <= n*n){
            for(int i = left; i < right; i++){
                maxtrix[top][i] = cnt++;
            }
            for(int i = top; i<buttom;i++){
                maxtrix[i][right] = cnt++;
            }
            if(left < right && top < buttom){
                for(int i = right; i > left; i--){
                    maxtrix[buttom][i] = cnt++;
                }
                for(int i = buttom; i > top; i--){
                    maxtrix[i][left] = cnt++;
                }
            }
            if(left == right && top == buttom){
                maxtrix[top][left] = cnt++;
            }
            left++;
            right--;
            top++;
            buttom--;
        }
        return maxtrix;
    }
}

File: leetcode/59.螺旋矩阵 II/59.py
class Solution:
    def generateMatrix(self, n: int) -> List[List[int]]:
        maxtrix = [[0] * n for _ in range(n)]
        left, right, top, buttom = 0, n-1, 0, n-1
        cnt = 1
        while left <= right and top <= buttom and cnt <= n*n:
            for i in range(left, right):
                maxtrix[top][i] = cnt
                cnt += 1
            for i in range(top, buttom):
                maxtrix[i][right] = cnt
                cnt += 1
            if left == right and top == buttom:
                maxtrix[top][left] = cnt
                cnt += 1
            if left < right and top < buttom:
                for i in range(right, left, -1):
                    maxtrix[buttom][i] = cnt
                    cnt += 1
                for i in range(buttom, top, -1):
                    maxtrix[i][left] = cnt
                    cnt += 1

            left += 1
            right -= 1
            top += 1
            buttom -= 1
        return maxtrix

        